apps/meteor/client/views/room/providers/hooks/useInstance.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { renderHook } from '@testing-library/react';
+
+import { useDepsMatch } from './useDepsMatch';
+import { useInstance } from './useInstance';
+
+jest.mock('./useDepsMatch', () => ({
+	useDepsMatch: jest.fn(),
+}));
+
+class MockChatMessages {
+	release() {
+		return 'released';
+	}
+}
+
+describe('useInstance', () => {
+	let factory: jest.Mock;
+
+	beforeEach(() => {
+		factory = jest.fn().mockReturnValue([{ instance: new MockChatMessages() }, jest.fn()]);
+	});
+
+	it('should create a new instance when dependencies change', () => {
+		(useDepsMatch as jest.Mock).mockReturnValue(false);
+
+		const { result, rerender } = renderHook(({ deps }) => useInstance(factory, deps), {
+			initialProps: { deps: ['initial-dep'] },
+			legacyRoot: true,
+		});
+
+		expect(factory).toHaveBeenCalledTimes(1);
+		expect(result.current).toEqual({ instance: new MockChatMessages() });
+
+		rerender({ deps: ['new-dep'] });
```

Comment:
I noticed you're mocking `useDepsMatch` return, as well as pushing new deps to the hook. I think you can remove the mock for `useDepsMatch`, just changing the array should be enough. I know you're also testing this mocked hook elsewhere, but since it's so simple of a hook I think it's better to just change the deps array instead of mocking the return value.

I recommend mocking already tested dependencies only if it decreases the test complexity enough to be justifiable. It's really a battle of how much you need to mock, sometimes mocking a return value is faster and less complex, some other times it could be the opposite, or hinder the test quality.